compiler/src/dotty/tools/dotc/ast/TreeInfo.scala

```diff
@@ -510,7 +510,7 @@ trait TypedTreeInfo extends TreeInfo[Type] { self: Trees.Instance[Type] =>
    */
   def constToLiteral(tree: Tree)(implicit ctx: Context): Tree = {
     val tree1 = ConstFold(tree)
-    tree1.tpe.widenTermRefExpr match {
+    tree1.tpe.widenTermRefExpr.dealias match {
```

Comment:
For later: maybe we should call `.normalized`, as `dealias` is just one of many possible normalization operations on types.

Reply:
Yes, I have that in #8843. But when I tried to normalize it here it did not work. Maybe I need to deeply dealias and then normalize.